lib/temporal/client/implementation.rb

```diff
@@ -196,16 +195,14 @@ def handle_query_workflow(input)
 
         if response.query_rejected
           status = Workflow::ExecutionStatus.from_raw(response.query_rejected.status)
-          # TODO: Replace with a specific error when implemented
-          raise Temporal::Error, "Query rejected, workflow status: #{status}"
+          raise Temporal::Error::QueryRejected, status
         end
 
         converter.from_payloads(response.query_result)&.first
       rescue Temporal::Bridge::Error => e
         # TODO: Raise a better error from the bridge
         if e.message.include?('unknown queryType')
```

Comment:
This is not a safe assumption that every lang's worker uses this text. You can use `INVALID_ARGUMENT` gRPC status code to determine that the worker failed the query, but you can't tell any more details than that. TypeScript (IMO erroneously) assumes that means not registered, but I think it can happen for any reason.

I think you should just check for `INVALID_ARGUMENT` gRPC status code and raise a `Temporal::Error::QueryFailed` with the message. (this is what Python does)

Reply:
Good point. I've made a note of this — https://github.com/temporalio/sdk-ruby/issues/71. I'll implement this along with properly parsing other Bridge errors (right now it's just a message without the status code decoded).

Reply:
Can you at least change `UnsupportedQuery` error to `QueryFailed`. There should not be an `UnsupportedQuery` error because you'll never be able to properly tell I don't believe.